### PR TITLE
Replace per-interconnect sockaddr formatting with generic implementation

### PR DESCRIPTION
**Pull request by danielgustafsson**
Each interconnect had its own (duplicate) function for converting a sockaddr_storage struct into a human readable string. Replace these with a shared implementation and convert it to use the upstream library function for just this purpose. We were already relying on this function in the case of an error on Solaris, a platform which isn't supported by Greenplum anyways.

I noticed this when poking a bit at commit 52c373720055e64f695ec82e84464803f583c24e which removed support for Solaris and HP-UX from Greenplum, and this function had workaround for a reported bug on Solaris. Seems we can simplify this code now.

### CODE REVIEW

**Comment by danielgustafsson** [comment]
> How does this work for IPV6? It seems `pg_getnameinfo_all() -> getnameinfo()` will always return nonzero return code `EAI_FAMILY`.

Why is that? `getnameinfo()` supports IPv6 on all modern platforms.. what am I missing?

File: src/backend/cdb/motion/ic_common.c [hunk 1]
```diff
@@ -887,3 +887,37 @@ interconnect_abort_callback(ResourceReleasePhase phase,
 		}
 	}
 }
+
+/*
+ * format_sockaddr
+ *			Format a sockaddr to a human readable string
+ *
+ * This function must be kept threadsafe, elog/ereport/palloc etc are not
+ * allowed to be called within this function context.
```
Comment (hlinnaka) [inline src/backend/cdb/motion/ic_common.c]: I would shorten this to just "..., elog/ereport/palloc etc. are not allowed within this function".

**Comment by danielgustafsson** [comment]
Pushed to master with comments fixed up as per review comment, thanks for review!
